plugins/catalog/src/alpha.tsx

```diff
@@ -97,27 +97,30 @@ export function createCatalogFilterExtension<
   configSchema?: PortableSchema<TConfig>;
   loader: (options: { config: TConfig }) => Promise<JSX.Element>;
 }) {
+  const id = `catalog.filter.${options.id}`;
+  const attachTo = { id: 'plugin.catalog.page.index', input: 'filters' };
+
   return createExtension({
-    id: `catalog.filter.${options.id}`,
-    attachTo: { id: 'plugin.catalog.page.index', input: 'filters' },
+    id,
+    attachTo,
     inputs: options.inputs ?? {},
     configSchema: options.configSchema,
     output: {
       element: coreExtensionData.reactElement,
     },
     factory({ bind, config, source }) {
-      const LazyComponent = React.lazy(() =>
+      const ExtensionComponent = React.lazy(() =>
         options
           .loader({ config })
           .then(element => ({ default: () => element })),
       );
 
       bind({
         element: (
-          <ExtensionBoundary source={source}>
-            <React.Suspense fallback={<Progress />}>
-              <LazyComponent />
-            </React.Suspense>
+          <ExtensionBoundary id={id} source={source}>
```

Comment:
We could consider passing the id to the factory to avoid the need for this refactor. Let's leave as is for now though and think about it again as we go over ID naming patterns imo

Reply:
Thanks for commenting, leaving as is for now 🙌🏻